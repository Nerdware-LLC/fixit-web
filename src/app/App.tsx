import { ApolloProvider } from "@apollo/client/react/context";
import { ErrorBoundary } from "@components/ErrorBoundary";
import { RootAppRouter } from "@routers/RootAppRouter";
import { DateTimeLocalizationProvider } from "./DateTimeLocalizationProvider";
import { GlobalStyles } from "./GlobalStyles";
import { PageLayoutContextProvider } from "./PageLayoutContext";
import { ThemeProvider } from "./ThemeProvider";
import { ToastContainer } from "./ToastContainer";
import { apolloClient } from "./apolloClient";

export const App = () => (
  <ApolloProvider client={apolloClient}>
    <ErrorBoundary>
      <PageLayoutContextProvider>
        <ThemeProvider>
          <DateTimeLocalizationProvider>
            <GlobalStyles />
            <RootAppRouter />
            <ToastContainer />
          </DateTimeLocalizationProvider>
        </ThemeProvider>
      </PageLayoutContextProvider>
    </ErrorBoundary>
  </ApolloProvider>
);
