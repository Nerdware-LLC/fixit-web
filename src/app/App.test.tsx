import renderer from "react-test-renderer";
import { it } from "vitest";
import { App } from "./App";

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
