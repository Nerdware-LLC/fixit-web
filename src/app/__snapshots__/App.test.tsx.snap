// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
[
  <div
    className="css-eao24d"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-xmpvs9-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": "7rem",
          "width": "7rem",
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={19.75}
          strokeWidth={4.5}
          style={{}}
        />
      </svg>
    </span>
  </div>,
  <div
    className="Toastify"
  />,
]
`;
